server/conn.go

```diff
@@ -144,9 +151,35 @@ func updateResponse(req *pdpb.Request, resp *pdpb.Response) {
 }
 
 func (c *conn) close() error {
+	if c.leaderConn != nil {
+		c.leaderConn.Close()
+	}
 	return errors.Trace(c.conn.Close())
 }
 
+func (c *conn) proxyRequest(msgID uint64, req *pdpb.Request) (*pdpb.Response, error) {
+	// Create a connection to leader.
+	if c.leaderConn == nil {
+		leader, err := c.s.GetLeader()
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		conn, err := rpcConnect(leader.GetAddr())
```

Comment:
When a leader become follower, it will close all connections, then this will failed and connect to the new leader next time.